### ISSUE TITLE
Skips first type parameter if it's a primitive type

### ISSUE THREAD

**Issue by antifuchs** [azriel91/tynm#1]
I'm trying to use `tynm` for [`governor`](https://crates.io/crates/governor), and noticed behavior that looks like a bug:

``` rust
assert_eq!(tynm::type_name::<std::collections::HashMap<u32, String>>(), "HashMap<u32, String>",);
```

fails:
```
  left: `"HashMap<, String>"`,
 right: `"HashMap<u32, String>"`', src/lib.rs:6:1
```

I guess somehow the first parameter gets dropped when the type is a builtin? It does work for `HashMap<String, String>`.

**Comment by azriel91** [comment]
ah, it's because it detects the type name based on case. Didn't think of primitive types :sweat_smile:.
Shall try and fix it now.

May not be so simple (`std::u32` is a module, `u32` is a type).
Also, just realized type names may also be `&[u32]`, `&str`, `[u32; 3]`, `[(u32, Something<T>); 3]`.

**Comment by antifuchs** [comment]
Huh! Yeah, I guess you are essentially writing a parser for type expressions in rust... wonder if there's something in, say, `syn` to help with that?

(update: yes, there is: [`parse_str`](https://docs.rs/syn/1.0.12/syn/fn.parse_str.html), available with syn's "parsing" feature)

**Comment by azriel91** [comment]
Haha, have halfway extended the `nom`my parser; switching to `syn` is tempting (haven't begun on `dyn` and `fn(T) -> U`).

**Comment by azriel91** [comment]
Published a fixed nom nom version in `0.1.1` :v:

```toml
tynm = "0.1.1"
```

**Comment by antifuchs** [comment]
Thanks!! This looks much better (:
